### PR TITLE
Ci add codeql

### PR DESCRIPTION
**Pull request by tacaswell**
## PR Summary

replaces #22446

### CODE REVIEW

**Comment by tacaswell** [comment]
I will squash this down once it is working.

**Comment by tacaswell** [comment]
I tried manually dismissing all of the warnings from `exten/` or `build/`, lets see if that reduces the noise?

I'm a bit unclear _where_ that state lives though.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'cpp', 'javascript', 'python' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
```
Comment (oscargus) [inline .github/workflows/codeql-analysis.yml:41]:
```suggestion
      uses: actions/checkout@v3
```
Minor thing put still...

**Review by oscargus — APPROVED** [review]
I think this looks good! Probably we still want someone more experienced than me to say something more insightful... (I am primarily experienced in browsing static code checking results, not setting them up...)

**Comment by tacaswell** [comment]
I would also like feedback from @QuLogic or @anntzer about the c++ changes to deal with the overflows (I did it two ways, is one better than the other? Is it important to be consistent here? should we have just pushed everything to bigger types?).

**Comment by anntzer** [comment]
It would probably be better to be consistent and use ssize_t everywhere, but let's not lose sleep over that for now.

**Comment by tacaswell** [comment]
Back to draft because the build does not work again...

File: src/_backend_agg.h [hunk 2]
```diff
@@ -1124,7 +1124,7 @@ class QuadMeshGenerator
 
     inline size_t num_paths() const
     {
-        return m_meshWidth * m_meshHeight;
+        return (size_t) m_meshWidth * m_meshHeight;
```
Comment (anntzer) [inline src/_backend_agg.h:1127]: py_adaptors.h defines num_paths as Py_ssize_t, perhaps better for consistency, though anything will work, of course.

**Comment by anntzer** [comment]
Nearly all the changes just hit the now deprecated to_string{,_argb}; just left an additional point re: signedness of num_paths, though it's not a big deal either way.

**Comment by QuLogic** [comment]
OK, this is now passing, though I will likely want to clean up the YAML. The JavaScript analysis takes ~6m; the C++/Python one takes ~17m. I think I can split C++/Python back up again now that it's working so that it'll finish quicker. Hopefully we don't end up with too many workflows running in parallel.

**Comment by QuLogic** [comment]
I squashed down everything, removed the C++ change and confirmed that CodeQL is still warning about it, so I've now added it back in along with a more explicit message about why the build broke earlier.

**Comment by tacaswell** [comment]
This is my PR so I can not approve it, but 👍🏻 from me.

Do we want to standardize the c++ any more or not worry about that right now?

**Comment by QuLogic** [comment]
I think since they're deprecated, it's fine to leave them slightly inconsistent since they'll be deleted.

**Pull request merged**